administrator/components/com_admin/script.php

```diff
@@ -2657,4 +2657,53 @@ private function cleanJoomlaCache()
 		$model->setState('client_id', 1);
 		$model->clean();
 	}
+
+	/**
+	 * Renames or removes incorrectly cased files.
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function fixFilenameCasing()
+	{
+		$files = array(
+			'libraries/src/Filesystem/Support/Stringcontroller.php' => 'libraries/src/Filesystem/Support/StringController.php',
+			'libraries/vendor/paragonie/sodium_compat/src/Core/Xsalsa20.php' => 'libraries/vendor/paragonie/sodium_compat/src/Core/XSalsa20.php',
+			'media/mod_languages/images/si_LK.gif' => 'media/mod_languages/images/si_lk.gif',
+		);
+
+		foreach ($files as $old => $expected)
+		{
+			$oldRealpath = realpath(JPATH_ROOT . '/' . $old);
+
+			// On Unix without incorrectly cased file.
+			if ($oldRealpath === false)
+			{
+				continue;
+			}
+
+			$oldBasename      = basename($oldRealpath);
+			$newRealpath      = realpath(JPATH_ROOT . '/' . $expected);
+			$newBasename      = basename($newRealpath);
```

Comment:
```suggestion
			$oldBasename      = pathinfo($oldRealpath, PATHINFO_BASENAME);
			$newRealpath      = realpath(JPATH_ROOT . '/' . $expected);
			$newBasename      = pathinfo($newRealpath, PATHINFO_BASENAME);
```
Maybe this will help with OSX?

Reply:
Nope. Same results

Reply:
Hmm, then we might have to do something with `glob`, check if we have 2 matching files for that base name, case insensitive, and only then delete the wrong file, and if only 1 match do the move. But today I'm done. Maybe I have some idea tomorrow.